clickhouse/datadog_checks/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,292 @@
+        rows = self.execute_query_raw(query.query, lambda: ', '.join(sorted(query.views)))
+
+        # Avoid repeated lookups
+        columns = query.columns
+        get_method = getattr
+
+        result = {}
+        for column, value in rows:
+            metric_data = columns.get(column)
+            if metric_data is None:
+                continue
+
+            for metric, method in metric_data:
+                if callable(method):
+                    method, computed_value = method(value)
+                    get_method(self, method)(metric, computed_value, tags=self._tags)
+                else:
+                    get_method(self, method)(metric, value, tags=self._tags)
+
+            result[column] = value
+
+        # Return all values for possible post-processing
+        return result
+
+    def execute_query_defined_schema(self, query):
+        rows = self.execute_query_raw(query.query, lambda: ', '.join(sorted(query.views)))
+
+        # Re-use column access map for efficiency
+        result = {}
+
+        for row in rows:
+            for column, value in zip(query.columns, row):
+                result[column] = value
+
+            yield result
+
+    def execute_query_raw(self, query, source, stream=False):
+        try:
+            if stream:
+                rows = self._client.execute_iter(query)
+
+                # Trigger query execution
+                try:
+                    first_row = next(rows)
+                except StopIteration:  # no cov
+                    return iter([])
+
+                return chain((first_row,), rows)
+            else:
+                return self._client.execute(query)
+        except Exception as e:
+            raise QueryExecutionError(self._error_sanitizer.clean(str(e)), source())
+
+    def create_connection(self):
+        try:
+            client = clickhouse_driver.Client(
+                host=self._server,
+                port=self._port,
+                user=self._user,
+                password=self._password,
+                database=self._db,
+                connect_timeout=self._connect_timeout,
+                send_receive_timeout=self._read_timeout,
+                sync_request_timeout=self._connect_timeout,
+                compression=self._compression,
+                secure=self._tls_verify,
+                # Don't pollute the Agent logs
+                settings={'calculate_text_stack_trace': False},
+                # Make every client unique for server logs
+                client_name='datadog-{}'.format(self.check_id),
+            )
+            client.connection.connect()
+        except Exception as e:
+            error = 'Unable to connect to ClickHouse: {}'.format(
+                self._error_sanitizer.clean(self._error_sanitizer.scrub(str(e)))
+            )
+            self.service_check(self.SERVICE_CHECK_CONNECT, self.CRITICAL, message=error, tags=self._tags)
+
+            raise_from(type(e)(error), None)
```

Comment:
So many questions about that line! Why rebuilding the error? Do we even know if they all take a single argument? Why use `raise_from` with `None` as argument, it won't provide a single additional detail?